src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.cs

```diff
@@ -277,54 +271,62 @@ public void Initialize(Solution solution)
                 throw new InvalidOperationException();
             }
 
-            // Create a connection to the DB and ensure it has tables for the types we care about. 
-            using var _ = GetPooledConnection(out var connection);
-
-            // Enable write-ahead logging to increase write performance by reducing amount of disk writes,
-            // by combining writes at checkpoint, salong with using sequential-only writes to populate the log.
-            // Also, WAL allows for relaxed ("normal") "synchronous" mode, see below.
-            connection.ExecuteCommand("pragma journal_mode=wal", throwOnError: false);
-
-            // Set "synchronous" mode to "normal" instead of default "full" to reduce the amount of buffer flushing syscalls,
-            // significantly reducing both the blocked time and the amount of context switches.
-            // When coupled with WAL, this (according to https://sqlite.org/pragma.html#pragma_synchronous and 
-            // https://www.sqlite.org/wal.html#performance_considerations) is unlikely to significantly affect durability,
-            // while significantly increasing performance, because buffer flushing is done for each checkpoint, instead of each
-            // transaction. While some writes can be lost, they are never reordered, and higher layers will recover from that.
-            connection.ExecuteCommand("pragma synchronous=normal", throwOnError: false);
-
-            // First, create all string tables in the main on-disk db.  These tables
-            // don't need to be in the write-cache as all string looks go to/from the
-            // main db.  This isn't a perf problem as we write the strings in bulk, 
-            // so there's no need for a write caching layer.  This also keeps consistency
-            // totally clear as there's only one source of truth.
-            connection.ExecuteCommand(
-$@"create table if not exists {StringInfoTableName}(
+            // This is our startup path.  Enqueue our work to the write queue (to ensure invariants about only writing
+            // while having access to it).  Then just block on that work.  The queue will be empty here, so this will
+            // not be costly to do.
```

Comment:
Does this run the body synchronously then or not? The risk here is still if the thread pool is saturated, we're going to end up blocking our thread on a thread that doesn't exist.

Reply:
So that is def a potential concern.  One way to alleviate that would be to revert this part and allow the initializer routine to get connection (i.e. by passing an explicit bool saying it's ok it's not using the scheduler pair).  

This woul dbe safe since all this runs with the startup of the service.  but would be a little less clean because we'd have to have that backdoor.  Curious about your thoughts here @jasonmalinowski

Reply:
I don't think I have any strong opinion either way. This current approach favors simplicity and correctness over _theoretical_ pref and I can't fault that.

Reply:
i personally think i'm going to change to not block here.  i do imagine we could see a tpl issue that would be highly unfortunate.